From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bouke Sybren Haarsma <boukehaarsma23@gmail.com>
Date: Sat, 27 Jul 2024 08:22:20 +0200
Subject: fixup aw87xxx.{c,h} warnings

This is deliberately an OGC-carried patch and must not be sent upstream.

(cherry picked from commit dac32102659fa0ae4f2692ed81810ceb834ed584)
Signed-off-by: Hijae Song <hijae@naver.com>
(cherry picked from commit 7610cab8a9d63d30930f8adde615347f7de8f947)

OGC-tag: NOT-FOR-UPSTREAM
---
 sound/soc/codecs/aw87xxx/aw87xxx.c | 1 +
 sound/soc/codecs/aw87xxx/aw87xxx.h | 5 +++++
 2 files changed, 6 insertions(+)

diff --git a/sound/soc/codecs/aw87xxx/aw87xxx.c b/sound/soc/codecs/aw87xxx/aw87xxx.c
index 688eae846e85f33cd2af0fc4b8a1cdcae9c04d64..960c0e780a6065d2bb7a46a3de2665615152293d 100644
--- a/sound/soc/codecs/aw87xxx/aw87xxx.c
+++ b/sound/soc/codecs/aw87xxx/aw87xxx.c
@@ -42,6 +42,7 @@
 #include <linux/hrtimer.h>
 #include <linux/ktime.h>
 #include <linux/kthread.h>
+#include <linux/vmalloc.h>
 #include <uapi/sound/asound.h>
 #include <sound/control.h>
 #include <sound/soc.h>
diff --git a/sound/soc/codecs/aw87xxx/aw87xxx.h b/sound/soc/codecs/aw87xxx/aw87xxx.h
index ed052226a003cea1a728d1be0f1bf7405f471561..4a613e4a8f123852372f1273e640de4cc2b00801 100644
--- a/sound/soc/codecs/aw87xxx/aw87xxx.h
+++ b/sound/soc/codecs/aw87xxx/aw87xxx.h
@@ -118,4 +118,9 @@ struct aw87xxx {
 int aw87xxx_update_profile(struct aw87xxx *aw87xxx, char *profile);
 int aw87xxx_update_profile_esd(struct aw87xxx *aw87xxx, char *profile);
 
+char *aw87xxx_show_current_profile(int dev_index);
+int aw87xxx_set_profile(int dev_index, char *profile);
+int aw87xxx_set_profile_by_id(int dev_index, int profile_id);
+int aw87xxx_add_codec_controls(void *codec);
+int aw87xxx_awrw_write(struct aw87xxx *aw87xxx, const char *buf, size_t count);
 #endif
